neps/nep-0452.md

```diff
@@ -0,0 +1,213 @@
+- The linkdrop claiming page connects to an existing NEAR Account.
+- Wallet will then use the `privKey1` which has access to call `claim` and `create_account_and_claim` in order to call `claim` on the contract.
+- The contract will transfer the funds to the existing account alongside any NFT or fungible tokens included in the linkdrop.
+
+
+## Reference Implementation
+
+Below are some references for linkdrop contracts. A specific implementation of this standard can be seen in [this pull request](https://github.com/near/near-linkdrop/pull/24).◊
+
+- [Link Drop Contract](https://github.com/near/near-linkdrop)
+
+- [Keypom Contract](https://github.com/keypom/keypom)
+
+
+## Security Implications
+
+1. Linkdrop Creation
+Linkdrop creation involves creating keypairs associated with mapping to assets such as $NEAR, FTs, NFTs and access keys. There are security implications with the access keys associated with the keypairs. These access keys should be function call access keys and be limited to specific functionality. For example, keypairs can be generated with function call access keys which have permission to only call `claim` and `create_account_and_claim`. Another important security implication of linkdrop creation is to ensure that only one key is mapped to a set of assets at any given time. Linkdrop creation can happen in real-time or prior to being received. The mass creation of linkdrops involves creating multiple keypairs associated with mapping to assets and access keys. The security implications mentioned above are increased with the mass creation of linkdrops.
+
+2. Linkdrop Key Management  
+Key management is a critical saftey component of linkdrops. The linkdrop creator should implement a key management strategy for linkdrop keys upon the receiver claiming the linkdrops assets. For example, one strategy may be to delete the linkdrop keypair in the same block the receiver claims the linkdrop assets, so a single linkdrop cannot be used multiple times. In another example, a linkdrop creator may allow for a linkdrop to be used X number of times, where they could track the number of calls to `claim` or `create_account_and_claim` and delete the keypair after X uses of the linkdrop. It is very important to implement your key management strategy in `claim` and `create_account_and_claim` as opposed to a callback function to ensure the linkdrop keys are not improperly used. Implementing key management in a callback has a security implication of such as 'double-spending' a linkdrop.
+
+## Future possibilities
+
+- Function call data types
+
+- Optional configurations added to KeyInfo which can include multi-usekeys, time-based claiming etc…
+
+- Standard process for how links connect to claim pages (i.e a standardized URL such as an app’s baseUrl/contractId=        [LINKDROP_CONTRACT]&secretKey=[SECRET_KEY]
+
+- Standard for deleting keys and refunding assets.
```

Comment:
Isn't it just a `claim` method, or do you want to keep track of a creator account id and allow the creator to claim back without the corresponding privKey?

Reply:
Yes - keep track of creator account and allow the creator to delete the keys without having to `claim` them